### PR TITLE
Refactor parquet tests

### PR DESCRIPTION
**Pull request by pdabre12**
Merged the ParquetWriterTestBase and ParquetReaderTestBase into ParquetTestBase for code re-usability.

### CODE REVIEW

**Comment by pdabre12** [comment]
@majetideepak Please review.

**Comment by majetideepak** [comment]
@duanmeng can you please help review this PR when you get a chance? Thanks.

File: velox/dwio/parquet/tests/ParquetTestBase.h [hunk 1]
```diff
@@ -16,27 +16,30 @@
...
-#include "velox/vector/tests/utils/VectorMaker.h"
+#include "velox/dwio/parquet/reader/PageReader.h"
+#include "velox/dwio/parquet/reader/ParquetReader.h"
+#include "velox/dwio/parquet/writer/Writer.h"
+#include "velox/exec/tests/utils/TempDirectoryPath.h"
+#include "velox/vector/tests/utils/VectorTestBase.h"
+#include "velox/vector/fuzzer/VectorFuzzer.h"
 
-#include <gtest/gtest.h>
+using namespace facebook::velox::test;
```
Comment (duanmeng) [inline velox/dwio/parquet/tests/ParquetTestBase.h:32]: Do not use `using namespace` in a header file.

File: velox/dwio/parquet/tests/ParquetTestBase.h [hunk 2]
```diff
@@ -16,27 +16,30 @@
...
 
 namespace facebook::velox::dwio::parquet {
```
Comment (duanmeng) [inline velox/dwio/parquet/tests/ParquetTestBase.h:34]: Is there any chance to unify the namespace to `facebook::velox::parquet`

File: velox/dwio/parquet/tests/ParquetTestBase.h [hunk 3]
```diff
@@ -126,16 +140,62 @@ class ParquetReaderTestBase : public testing::Test {
...
-  std::unique_ptr<test::VectorMaker> vectorMaker_{
-      std::make_unique<test::VectorMaker>(pool_.get())};
+  std::shared_ptr<memory::MemoryPool> rootPool_;
+  std::shared_ptr<memory::MemoryPool> leafPool_;
+  std::shared_ptr<exec::test::TempDirectoryPath> tempPath_;
+  static constexpr uint64_t kRowsInRowGroup = 10'000;
+  static constexpr uint64_t kBytesInRowGroup = 128 * 1'024 * 1'024;
 };
-} // namespace facebook::velox::dwio::parquet
+} // namespace facebook::velox::dwio::parquet
```
Comment (duanmeng) [inline velox/dwio/parquet/tests/ParquetTestBase.h:201]: Add an empty line at the end of the file.

File: velox/dwio/parquet/tests/reader/CMakeLists.txt [hunk 4]
```diff
@@ -19,7 +19,7 @@ add_test(
   WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
 target_link_libraries(
   velox_dwio_parquet_reader_test velox_dwio_native_parquet_reader
-  velox_link_libs ${TEST_LINK_LIBS})
+  velox_link_libs velox_exec_test_lib velox_temp_path ${TEST_LINK_LIBS})
```
Comment (duanmeng) [inline velox/dwio/parquet/tests/reader/CMakeLists.txt:22]: could we add `velox_exec_test_lib velox_temp_path` to `${TEST_LINK_LIBS}`?

File: velox/dwio/parquet/tests/reader/CMakeLists.txt [hunk 5]
```diff
@@ -28,7 +28,7 @@ add_test(
   WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
 target_link_libraries(
   velox_dwio_parquet_page_reader_test velox_dwio_native_parquet_reader
-  velox_link_libs ${TEST_LINK_LIBS})
+  velox_link_libs velox_exec_test_lib velox_temp_path ${TEST_LINK_LIBS})
```
Comment (duanmeng) [inline velox/dwio/parquet/tests/reader/CMakeLists.txt:31]: ditto

File: velox/dwio/parquet/tests/writer/SinkTests.cpp [hunk 6]
```diff
@@ -84,4 +70,4 @@ TEST_F(SinkTest, abort) {
   // Abort would not flush.
   writer->abort();
   ASSERT_EQ(size, fs::file_size(filePath));
-}
+}
```
Comment (duanmeng) [inline velox/dwio/parquet/tests/writer/SinkTests.cpp:73]: empty line

**Review by duanmeng — COMMENTED** [review]
Nice refactor. Looks good % some minor comments.

**Comment by pdabre12** [comment]
@duanmeng Thank you for the review. Made all the suggested changes.

File: velox/dwio/parquet/tests/writer/SinkTests.cpp (hunk 6 above)
Comment (pdabre12) [reply velox/dwio/parquet/tests/writer/SinkTests.cpp]: Did you mean add an empty line? format-check is failing because of the extra empty line.

**Comment by duanmeng** [comment]
@pdabre12 You can run `python scripts/check.py format commit --fix` locally before you push, which would check and fix your format issue locally, this would save you lots of time. If your local branch is main, you could just run `make format --fix`.

File: velox/dwio/parquet/tests/writer/SinkTests.cpp (hunk 6 above)
Comment (duanmeng) [reply velox/dwio/parquet/tests/writer/SinkTests.cpp]: I mean add an empty line at the end of the file.

Comment (pdabre12) [reply velox/dwio/parquet/tests/writer/SinkTests.cpp]: yes , I did that, format-check throws the empty line as an error.

**Comment by pdabre12** [comment]
> @pdabre12 You can run `python scripts/check.py format commit --fix` locally before you push, which would check and fix your format issue locally, this would save you lots of time. If your local branch is main, you could just run `make format --fix`.

Got it, thank you!

File: velox/dwio/parquet/tests/writer/SinkTests.cpp (hunk 6 above)
Comment (duanmeng) [reply velox/dwio/parquet/tests/writer/SinkTests.cpp]: It's weird. Let's run `python scripts/check.py format commit --fix` and let the checker fix it automatically.

Comment (duanmeng) [reply velox/dwio/parquet/tests/writer/SinkTests.cpp]: PS: This should be run after committing all your changes locally.

**Comment by duanmeng** [comment]
@pdabre12 Let's update the PR title by removing the `[native]` prefix. I believe this prefix should be used in `prestodb/presto` only, since the whole Velox project is **native**.

File: velox/dwio/parquet/tests/CMakeLists.txt [hunk 7]
```diff
@@ -19,6 +19,8 @@ set(TEST_LINK_LIBS
     gtest_main
     gmock
     gflags::gflags
+    velox_exec_test_lib
```
Comment (majetideepak) [inline velox/dwio/parquet/tests/CMakeLists.txt:22]: nitpicking: move all `velox_*` libraries together on top

File: velox/dwio/parquet/tests/ParquetTestBase.h [hunk 8]
```diff
@@ -126,16 +138,63 @@ class ParquetReaderTestBase : public testing::Test {
...
         "velox/dwio/parquet/tests/reader", "../examples/" + fileName);
   }
 
-  std::shared_ptr<memory::MemoryPool> pool_{memory::addDefaultLeafMemoryPool()};
-  std::unique_ptr<test::VectorMaker> vectorMaker_{
-      std::make_unique<test::VectorMaker>(pool_.get())};
+  std::shared_ptr<memory::MemoryPool> rootPool_;
+  std::shared_ptr<memory::MemoryPool> leafPool_;
+  std::shared_ptr<exec::test::TempDirectoryPath> tempPath_;
+  static constexpr uint64_t kRowsInRowGroup = 10'000;
```
Comment (majetideepak) [inline velox/dwio/parquet/tests/ParquetTestBase.h:196]: static variables should appear first in the list. Move them above `rootPool_`

File: velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp [hunk 9]
```diff
@@ -624,9 +641,9 @@ TEST_F(ParquetReaderTest, preloadSmallFile) {
...
 
   constexpr int kBatchSize = 10;
-  auto result = BaseVector::create(sampleSchema(), 1, pool_.get());
+  auto result = BaseVector::create(sampleSchema(), 1, leafPool_.get());
   while (rowReader->next(kBatchSize, result)) {
     // Check no duplicate reads.
     ASSERT_EQ(file->bytesRead(), 0);
   }
-}
+}
```
Comment (majetideepak) [inline velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp:649]: add a new line here

**Review by majetideepak — COMMENTED** [review]
@pdabre12 Few minor comments

**Comment by majetideepak** [comment]
@pdabre12 Please rebase with the main branch.

**Comment by pdabre12** [comment]
@majetideepak @duanmeng  Please review. 
Was facing some namespace issues but resolved them , however adding an extra line in both the `ParquetReaderTest.cpp` and the `SinkTests.cpp` fails the format check for me. I used the python scripts/check.py format commit --fix script locally to fix the format issues.

**Comment by duanmeng** [comment]
@pdabre12 LGTM, the ci build failed, could you rebase?

**Comment by pdabre12** [comment]
@duanmeng Please review now. Fixed build issues.

**Comment by pdabre12** [comment]
Rebasing again to fix conflicts.

**Comment by pdabre12** [comment]
@majetideepak Please review.

**Review by duanmeng — APPROVED** [review]
LGTM, thanks.

File: velox/dwio/parquet/tests/ParquetTestBase.h [hunk 10]
```diff
@@ -126,16 +138,63 @@ class ParquetReaderTestBase : public testing::Test {
...
+      facebook::velox::common::CompressionKind compressionKind) {
+    facebook::velox::parquet::WriterOptions options;
+    options.memoryPool = rootPool_.get();
+    options.flushPolicyFactory = flushPolicy;
+    options.compression = compressionKind;
+    return std::make_unique<facebook::velox::parquet::Writer>(std::move(sink), options);
+  }
+
+  std::pair<std::unique_ptr<facebook::velox::parquet::Writer>, dwio::common::FileSink*>
+  createWriterWithSinkPtr(
```
Comment (majetideepak) [inline velox/dwio/parquet/tests/ParquetTestBase.h:166]: This API is odd. I see that this is used by `SinkTest.cpp`. I think we should simply inline this function.

**Review by majetideepak — COMMENTED** [review]
@pdabre12 a minor comment. Changes look good!

File: velox/dwio/parquet/tests/ParquetTestBase.h (hunk 10 above)
Comment (pdabre12) [reply velox/dwio/parquet/tests/ParquetTestBase.h:166]: This API will further also be used by the Parquet Writer Tests as well.

Comment (majetideepak) [reply velox/dwio/parquet/tests/ParquetTestBase.h:166]: There is already a `createSink` and `createWriter` API. Users can use them without much effort. Let's remove this API.
```
auto sink = createSink(filePath);
auto sinkPtr = sink.get();
auto writer = createWriter(std::move(sink), ...);
```

Comment (pdabre12) [reply velox/dwio/parquet/tests/ParquetTestBase.h]: Thanks @majetideepak. Made the necessary changes !

**Review by majetideepak — APPROVED** [review]
Thanks, @pdabre12

**Comment by majetideepak** [comment]
@pedroerp can you help review this PR? Thanks.

**Pull request closed without merging**